### ISSUE TITLE
Release 1.x.x acceptance criteria

### ISSUE THREAD

**Issue by mfendeksilverstripe** [maxime-rainville/silverstripe-anyfield#28]
# Release 1.x.x acceptance criteria

We are aiming to get this module to a stable `1.x.x` release state. This issue covers the acceptance criteria for that.

This module is valuable as it provides expanded options for Elemental inline editor, specifically a much needed GridField replacement. Note that this does not provide all capabilities that GridField used to have but it does cover quite a significant chunk of it.

What we're currently looking it as item management of models nested under content blocks but in the future the capability of this module can be enhanced further.

Notable features:

* `has_many` relation support for item management (created, read, update, delete)
* sorting

Ideally, we can put together a roadmap for this module with the aim to have a stable release by September 1st 2024.

![Screenshot 2023-12-14 at 1 38 25 PM](https://github.com/maxime-rainville/silverstripe-anyfield/assets/26395487/b42363d9-9fc0-452e-a6ce-73ecd53b9564)

![Screenshot 2023-12-14 at 1 39 53 PM](https://github.com/maxime-rainville/silverstripe-anyfield/assets/26395487/f0b44ea9-d04f-4a41-9788-90f229f7c257)

## Must have AC

These requirements are mandatory.

* https://github.com/maxime-rainville/silverstripe-anyfield/issues/31 (this is MVP)
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/11 (this is MVP)
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/16
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/12
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/30
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/10
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/9

## Nice to have AC

These requirements are optional for the initial release but should be delivered in future minor versions.

* https://github.com/maxime-rainville/silverstripe-anyfield/issues/34
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/24
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/17
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/23
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/19
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/18
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/14
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/8
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/29
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/15
* https://github.com/maxime-rainville/silverstripe-anyfield/issues/7

## Notes

* [done] before starting implementation check if we should rebuild this on top of `LinkField v4` as the current implementation is built on top of `LinkField v3`
* the decision was made to keep the current implementation based on `LinkField v3` as we want to get stable release out without overcommitting on the effort - once this module becomes officially supported we can create a new major version based on `LinkField v4`

**Comment by mfendeksilverstripe** [comment]
FYI: @maxime-rainville

**Comment by maxime-rainville** [comment]
In my mind the following "must have" are more in the "nice to have" category:
- https://github.com/maxime-rainville/silverstripe-anyfield/issues/17
- https://github.com/maxime-rainville/silverstripe-anyfield/issues/15

Some of the cards that were still open, were actually completed already, so I close those.

**Comment by mfendeksilverstripe** [comment]
Next steps: we need to make sure this module works with LinkField 4 and ideally make the validation work the same way as LinkField.

**Comment by sunnysideup** [comment]
Has anything happened in this space?
